gammapy/maps/axes.py

```diff
@@ -339,6 +339,21 @@ def to_node_type(self, node_type):
                 node_type=node_type,
                 unit=self.unit,
             )
+    
+    def rename(self, new_name):
+        """Rename the axis.
+    
+        Parameters
+        ----------
+        new_name : str
+            The new name for the axis.
+
+        Returns
+        -------
+        axis : `~gammapy.maps.MapAxis`
+            Renamed MapAxis
+        """
```

Comment:
We could maybe add a check to `MapAxis.__init__`, that `name` is actually as string...not sure we have this.